### DISCUSSION TITLE
Fatal: repository not found

### DISCUSSION THREAD

**Discussion by Marvelous-Software** [community/community#22000] [New to GitHub]
<p>Hello</p>
<p>I cant update my repository.  The message I get is:</p>
<p>fatal: repository ‘<a href="https://github.com/aceofwings/RotairERP.git/" rel="nofollow noopener">https://github.com/aceofwings/RotairERP.git/</a>’ not found</p>
<p>when I enter git remote -v I get</p>
<p>origin  <a href="https://github.com/aceofwings/RotairERP.git" rel="nofollow noopener">https://github.com/aceofwings/RotairERP.git</a> (fetch)<br>
origin  <a href="https://github.com/aceofwings/RotairERP.git" rel="nofollow noopener">https://github.com/aceofwings/RotairERP.git</a> (push)</p>
<p>I havent been using git that long.  I dont know what it means when one command cant find the repository another command reports.</p>
<p>I did log into git with a different accout then sqitch back.  I suspect that some part didnt switch over while another part did.  Just my best guess.  Before I spend hours trying to figure out this pecularity I thought best to see if someone with more knoweldge of git may know what this means.</p>
<p>I am using windows 10 and one repository is private and the other is public, if that matters.</p>
<p>Thanks</p>

**Comment by seveas** [comment, marked as answer]
<p>That repository is probably private. Try: </p>
<pre><code>git remote set-url origin https://YOUR_GITHUB_USER@github.com/aceofwings/RotairERP.git
</code></pre>

**Reply by AZvkm380** [reply]
Am getting this error "fatal: not a git repository (or any of the parent directories): .git" when try this out

**Comment by Marvelous-Software** [comment]
<p>Yes, it is private.</p>
<p>C:\Code86\RotairERP&gt;git remote set-url origin <a href="https://Marvelous-Software@github.com/aceofwings/RotairERP.git" rel="nofollow noopener">https://Marvelous-Software@github.com/aceofwings/RotairERP.git</a></p>
<p>C:\Code86\RotairERP&gt;git push origin master<br>
remote: Repository not found.<br>
fatal: repository ‘<a href="https://Marvelous-Software@github.com/aceofwings/RotairERP.git/" rel="nofollow noopener">https://Marvelous-Software@github.com/aceofwings/RotairERP.git/</a>’ not found</p>
<p>And yes I do have rights to the repository:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/11b5284c44e40122c37dcbd1346875dfd37d61c3.png" data-download-href="/uploads/short-url/2wEfbEp3z09xLJOewQ3u0PAco7N.png?dl=1" title="github.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/27378699/181068158-47bd1312-b094-47f3-800d-d4b5ca195f18.png" alt="github.png" data-base62-sha1="2wEfbEp3z09xLJOewQ3u0PAco7N" width="690" height="484" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/1/11b5284c44e40122c37dcbd1346875dfd37d61c3_2_690x484.png, https://aws1.discourse-cdn.com/github/optimized/2X/1/11b5284c44e40122c37dcbd1346875dfd37d61c3_2_1035x726.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/1/11b5284c44e40122c37dcbd1346875dfd37d61c3.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/11b5284c44e40122c37dcbd1346875dfd37d61c3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">github.png</span><span class="informations">1055×741 45.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Is there any documentation for the error messages?  Ive seen the book, <a href="https://git-scm.com/book/en/v2" rel="nofollow noopener">https://git-scm.com/book/en/v2</a></p>
<p>But that is only useful when everything is working.  I cant find any trouble shooting guides.  </p>
<p>Any assistance is appreciated.</p>

**Reply by dankz-net** [reply]
Nice man, me help very!  :D

**Comment by lee-dohm** [comment]
<p>Has <span class="mention">@aceofwings</span> added you as a collaborator on that repository? Since it is private, if they haven’t done that, you won’t be able to access the repository. Beyond that, since it is a private repository you should probably contact private support at <a href="https://github.com/contact" rel="nofollow noopener">https://github.com/contact</a>. They will have the access to take a look at what might be going wrong.</p>

**Comment by Marvelous-Software** [comment]
<p>Thanks for your reply.</p>
<p>OK, I’ll try that.  But with several days downtime, a couple of hours trying to fix the issue github appears to be too expensive as far as lost time goes to be worth using.  Unless your suggestion saves the day.</p>
<p>thanks</p>

**Comment by Marvelous-Software** [comment]
<p>This can be closed.  Since we cant use Github we have decided to use an alternate.</p>

**Comment by skipstevenson** [comment]
<p>I have a similar issue to Marvelous-Software.  I created/initialized a local repo (“testrepo”) in Bash/Git, added a file and did an initial commit.  I tried running this command:</p>
<p>git remote add origin <a href="https://github.com/" rel="nofollow noopener">https://github.com/</a>&lt;my-github-username&gt;testrepo.git</p>
<p>as well as one similar to what you suggested:</p>
<p>git remote set-url origin <a href="https://github.com" rel="nofollow noopener">https://github.com</a> /&lt;my-github-username&gt;testrepo.git</p>
<p>In both cases, the command git remote --verbose returned results that the above url (minus the .git at the end) existed for fetch and pull.</p>
<p>However, when I go to my github account, the aforementioned repo doesn’t exist.</p>
<p>Any suggestions?</p>

**Comment by lee-dohm** [comment]
<p><a class="mention" href="/skipstevenson">@skipstevenson</a> when you create a repo locally, GitHub has no way of knowing you did that. So you have to tell GitHub to create a repo on GitHub for you. You can do this in a few different ways, the simplest one is to go to <a href="https://github.com/new" rel="nofollow noopener">https://github.com/new</a> and follow the instructions. Once you’ve done that, then the remote connection you’ve defined already will work.</p>
<p>I hope that helps!</p>

**Comment by tharidudifferent** [comment]
<p>go to <a href="https://github.community/t5/How-to-use-Git-and-GitHub/git-clone-is-not-working-for-a-private-repo/m-p/2519#M816" rel="nofollow noopener">this url</a>. It worked for me.</p>

**Comment by TKGordon** [comment]
<p>Thank you, this worked! Trying to set up GitHub for the first time, and I kept getting the fatal : repository not found error, despite the repository being public in GitHub. </p>
<p>Used the command you provided, followed by push, and success.</p>

**Comment by electrocoder** [comment]
<p>The same problem is in my private repo.</p>

**Reply by faizankachchi** [reply]
same bro

**Comment by theoph21** [comment]
<p>This trick saved me big time.</p>
<p>It turned out that I had old session state:</p>
<ol>
<li>Open Control Panel from the Start menu</li>
<li>Select User Accounts</li>
<li>Select “Manage your credentials” in the left hand menu</li>
<li>Delete any credentials related to Git or GitHub</li>
</ol>
<p>Once I did this, it started working again.</p>

**Reply by HanObi83** [reply]
Thank you! This was exactly what I needed to fix this issue!

**Reply by nicKondratiev** [reply]
Thank you so much :)

**Reply by Sumanmali019** [reply]
How should i do it in mac

**Reply by AlbeiroJ** [reply]
Thank you!!!

**Reply by sjavidahamed** [reply]
Thank you..it worked

**Reply by Dante2805** [reply]
It was very helpful. Thank you so much. Your answer should be the first one in this discussion :3

**Reply by lsea-cecilia** [reply]
Thank you so much!!!

**Comment by learn-devops-fast** [comment]
<p>I put together a list of five different reasons why this error might happen over on TSS. You might find this helpful.</p>
<p><a href="https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/Five-ways-to-fix-Gits-fatal-repository-not-found-error" rel="nofollow noopener">https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/Five-ways-to-fix-Gits-fatal-repository-not-found-error</a></p>

**Comment by hcone5006** [comment]
<p>This worked for me, thank you! I have been using Source Control on VS Code for ages and it suddenly stopped working but this fixed it… cheers</p>

**Comment by UTfrank** [comment]
<p>I just tried </p>
<pre><code>git remote set-url origin https://YOUR_GITHUB_USER@github.com/aceofwings/RotairERP.git
</code></pre>
<p>and it didn’t work </p>

**Comment by elipeters** [comment]
<p>Works great!</p>
<p>Then create a <a href="https://github.com/settings/tokens" rel="nofollow noopener">Personal Access Token</a> with repo &amp; read/write permissions (or whatever your use case is) as the password.</p>

**Comment by sinclairjaza** [comment]
<p>Thanks <a class="mention" href="/skipstevenson">@skipstevenson</a>  for this solution. It work for me</p>

**Comment by glestwid** [comment]
<p>Trying to push my the very first commit into my private repo:</p>
<p>C:\TMP\New folder&gt;git remote add origin <a href="https://github.com/glestwid/tcc.git" rel="nofollow noopener">https://github.com/glestwid/tcc.git</a><br>
fatal: remote origin already exists.</p>
<p>C:\TMP\New folder&gt;git remote --verbose set-url origin <a href="https://github.com/glestwid/tcc.git" rel="nofollow noopener">https://github.com/glestwid/tcc.git</a></p>
<p>C:\TMP\New folder&gt;git push origin master<br>
remote: Repository not found.<br>
fatal: repository ‘<a href="https://github.com/glestwid/tcc.git/" rel="nofollow noopener">https://github.com/glestwid/tcc.git/</a>’ not found</p>
<p>C:\TMP\New folder&gt;</p>
<p>I do see the repo on my page at Github but at the same time git nither asks me regarding credentials nor gives me an access. What did i do wrong?</p>

**Comment by lee-dohm** [comment]
<p>Is the name of the repo <code>tcc</code>? Or <code>tcc.git</code>? If it is <code>tcc.git</code> then the remote you listed could be ambiguous.</p>

**Comment by valeriiisotnyk** [comment]
Just to add. If you use the old password/token for the new private repo, you need to create a new token.
Also, `git credential-manager uninstall` before push may help

**Comment by DAAAAAAAAAAAAAAAAN** [comment]
In my case, this error was caused by authenticating with an SSH key which was uploaded to my GitHub account through the GitKraken client. I will let the GitHub support response explain the issue:

> Thank you for contacting GitHub support. It looks like the only SSH key that is associated with your user account was generated by the GitKraken OAuth application. In order to use this SSH key in the <redacted> organization, you would need to [request organization approval](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/requesting-organization-approval-for-oauth-apps) for GitKraken and have it be granted. Otherwise, you can create a [new SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh) that is not associated with an OAuth application so no organization approval is required.

I generated a new key and added it to github according to their documentation, then configured my GitKraken client to use the local ssh-agent. From then it worked.

They further explained that right now, there is no good way to learn this information yourself. Best of luck, developer from the future!

**Comment by aceofwings** [comment]
Simple.

I did not give this man access.

**Comment by Hz-Yan** [comment]
Since it is a private repo, you have to run `gh auth login`
